services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java

```diff
@@ -238,4 +231,44 @@ public Integer getUsersCount() {
 
         return session.users().getUsersCount(realm);
     }
+
+    private List<UserRepresentation> searchForUser(Map<String, String> attributes, RealmModel realm, UserPermissionEvaluator usersEvaluator, Boolean briefRepresentation, Integer firstResult, Integer maxResults, Boolean includeServiceAccounts) {
+        if (!usersEvaluator.canView()) {
+            List<String> groupModels = auth.groups().canViewMembers();
+
+            if (!groupModels.isEmpty()) {
+                try {
+                    attributes.put(UserModel.GROUPS, JsonSerialization.writeValueAsPrettyString(groupModels));
+                } catch (IOException cause) {
+                    throw new RuntimeException("Failed to create groups filter", cause);
+                }
+            }
+        }
+
+        attributes.put(UserModel.INCLUDE_SERVICE_ACCOUNT, includeServiceAccounts.toString());
+
+        List<UserModel> userModels = session.users().searchForUser(attributes, realm, firstResult, maxResults);
```

Comment:
This triggers the lazy-loading of all user associations (attributes, credentials) so the effect of the briefRepresentation-change from @hmlnarik in L252 is lost. The loading is triggered in JpaUserProvider.java L784 calling UserModel getUserById(String id, RealmModel realm) that caches the user.